### PR TITLE
feat: update Ruby recorder code

### PR DESCRIPTION
**Pull request by eglitise**
This PR updates the recorder-generated code for the Ruby lib core client.

* Fix `uiautomator` locator strategy
* Do not add quotes around dictionary keys
* Fix `executeScript`
* Add `activateApp` and `terminateApp`
* Change many methods deprecated in `ruby_lib_core` 7.3 to their mobile extension equivalents
  * Certain mobile extensions like `mobile: setConnectivity` first require UI changes
* Reorder and group driver command implementations

### CODE REVIEW

**Review by jlipps — APPROVED** [review]

File: app/renderer/lib/client-frameworks/ruby.js [hunk 1]
```diff
@@ -298,29 +337,33 @@ driver.quit`;
...
   }
 
   codeFor_getContexts () {
-    return `driver.available_contexts`;
+    return `contexts = driver.available_contexts`;
   }
 
   codeFor_switchContext (varNameIgnore, varIndexIgnore, name) {
-    return `driver.context = '${name}'`;
+    return `driver.set_context '${name}'`;
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:366]: Both work as https://github.com/appium/ruby_lib_core/blob/1d438743e7b88227dfe0d1477955da61377cb401/lib/appium_lib_core/common/base/driver.rb#L472-L473
```suggestion
    return `driver.context = '${name}'`;
```

`driver.context =` is more Ruby friendly syntax, so here can keep the previous one

File: app/renderer/lib/client-frameworks/ruby.js [hunk 2]
```diff
@@ -298,29 +337,33 @@ driver.quit`;
...
+    return `current_url = driver.get_current_url`;
+  }
+
+  codeFor_back () {
+    return `driver.back`;
   }
 
   codeFor_forward () {
-    return `driver.navigate().forward`;
+    return `driver.forward`;
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:348]:
```suggestion
    return `driver.navigate.forward`;
```

File: app/renderer/lib/client-frameworks/ruby.js [hunk 3]
```diff
@@ -298,29 +337,33 @@ driver.quit`;
...
   }
 
   codeFor_refresh () {
-    return `driver.navigate().refresh`;
+    return `driver.refresh`;
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:352]:
```suggestion
    return `driver.navigate.refresh`;
```

File: app/renderer/lib/client-frameworks/ruby.js [hunk 4]
```diff
@@ -272,23 +319,15 @@ driver.quit`;
...
-      return settings;
-    } catch (e) {
-      return `// Could not parse: ${settingsJson}`;
-    }
+    return `driver.update_settings ${this.getRubyVal(settingsJson)}`;
   }
 
   codeFor_getSettings () {
-    return `settings = driver.settings.get`;
+    return `settings = driver.get_settings`;
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:330]:
```suggestion
    return `settings = driver.settings.get`;
```

File: app/renderer/lib/client-frameworks/ruby.js [hunk 5]
```diff
@@ -7,29 +7,40 @@ class RubyFramework extends Framework {
...
 caps = {}
 ${capStr}
-opts = {
-    server_url: "${this.serverUrl}"
+
+appium_lib_opts = {
+  server_url: "${this.serverUrl}"
 }
-driver = Appium::Core.for({caps: caps, appium_lib: opts}).start_driver
+driver = Appium::Core.for({caps: caps, appium_lib: appium_lib_opts}).start_driver
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:38]:
```suggestion
core = Appium::Core.for url: "${this.serverUrl}", caps: caps
driver = core.start_driver
```

File: app/renderer/lib/client-frameworks/ruby.js [hunk 6]
```diff
@@ -7,29 +7,40 @@ class RubyFramework extends Framework {
...
 require 'appium_lib_core'
 
 caps = {}
 ${capStr}
-opts = {
-    server_url: "${this.serverUrl}"
+
+appium_lib_opts = {
+  server_url: "${this.serverUrl}"
 }
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:37]: This appium_lib_opts can be removed

File: app/renderer/lib/client-frameworks/ruby.js [hunk 7]
```diff
@@ -135,36 +163,79 @@ driver.quit`;
...
   }
 
-  codeFor_pullFolder (varNameIgnore, varIndexIgnore, folderToPullFrom) {
-    return `driver.pull_folder '${folderToPullFrom}'`;
+  codeFor_isLocked () {
+    return `is_locked = ${this.codeFor_executeScriptNoArgs('mobile: isLocked')}`;
+  }
+
+  codeFor_rotateDevice () {
+    return `# Not supported: rotateDevice`;
```
Comment (KazuCocoa) [inline app/renderer/lib/client-frameworks/ruby.js:200]: Not for this issue itself. but maybe we no longer function this method itself.
https://github.com/search?q=org%3Aappium%20mobileRotation&type=code

**Review by KazuCocoa — APPROVED** [review]
thank you!

File: app/renderer/lib/client-frameworks/ruby.js (hunk 7 above)
Comment (KazuCocoa) [reply app/renderer/lib/client-frameworks/ruby.js:200]: -> https://github.com/appium/appium/issues/19378

Comment (eglitise) [reply app/renderer/lib/client-frameworks/ruby.js:200]: Yes, they should likely be replaced with `getRotation`/`setRotation`. Not sure if the current drivers implement it though.

**Pull request merged**